docs/api/proposed_endpoints.md

```diff
@@ -0,0 +1,186 @@
+[TOC]
+
+# Posts related actions
+
+## Create a post
+
+Create a post of contents.
+
+```
+POST api/posts
+```
+
+**Paramaters**
+
+Name         | Description
+-------------|--------------------------------------
+id           | identity of the post
+posted_by    | user who is posting the content
+posted_at    | social media ploatform to post
+posted_time  | time at which post should be posted
+content_type | type of content, text or image
+content      | content of the post
+image        | image in the post
+is_approved  | whether the post is approved or not
+is_posted    | whether the post is posted or not
+
+**Request**
+```json
+{
+    "id": "724",
+    "posted_by": "John Micky",
+    "posted_at": "Twitter",
+    "posted_time": "",
```

Comment:
Add a time date time here in ISO 8601 format.

Reply:
But `posted_time` is the time when the post should be posted. Therefore, while creating the post `posted_time` is left empty so that later on, once the post is created an admin or moderator can update the post to set this `posted_time`.

Reply:
@GeekyShacklebolt when we are proposing an endpoint, it's important to provide all the necessary information in order to understand how API is working. We can make the `posted_time` hold null or blank value, but for the sake of documentation the API should be complete.

Reply:
Let's rename that to `submitted_at`, when the user actually submitted the request for the post and then `posted_at` refers to the actual time when it was posted.

Makes sense?

Reply:
See, how naming makes a big difference :)